gci/src/main/java/com/danielfireman/gci/Sampler.java

```diff
@@ -0,0 +1,49 @@
+package com.danielfireman.gci;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class Sampler {
+	// Default sample rate should be fairly small, so big requests get checked up quickly.
+	private final int DEFAULT_SAMPLE_RATE = 10; // TODO(David) Update this value, if needed
+	// Max sample rate can not be very big because of peaks.
+	// The algorithm is fairly conservative, but we never know.
+	private final int MAX_SAMPLE_RATE = 30; // TODO(David) Update this value, if needed
+	
+	private int next;
+	private int[] pastSampleRates;
+	private AtomicInteger currentSampleSize; 
+
+	public Sampler(int historySize) {
+		currentSampleSize = new AtomicInteger(DEFAULT_SAMPLE_RATE); 
+		pastSampleRates = new int[historySize];
+		for (int i = 0; i < historySize; i++) { 
+			pastSampleRates[i] = Integer.MAX_VALUE;
+		}
+	}
+	
+	public int getCurrentSampleSize() {
+		return currentSampleSize.get();
+	}
+	
+	public void updateSampler(int lastFinished) { 
```

Comment:
javadoc?

Reply:
I was intended to write javadoc after have done all implementation. What do you think?

Reply:
That's fine too. Write the javadoc when writing the unit tests.